src/lib.rs

```diff
@@ -199,20 +210,26 @@ impl DfuAsyncIo for DfuNusb {
         value: u16,
         buffer: &[u8],
     ) -> Result<Self::Write, Self::Error> {
+        let interface = self.interface.as_ref().ok_or(std::io::Error::new(
+            std::io::ErrorKind::NotConnected,
+            "Interface closed",
+        ))?;
+
         let (control_type, recipient) = split_request_type(request_type);
         let req = ControlOut {
             control_type,
             recipient,
             request,
             value,
-            index: self.interface.interface_number() as u16,
+            index: interface.interface_number() as u16,
             data: buffer,
         };
-        let r = self.interface.control_out(req).await.into_result()?;
+        let r = interface.control_out(req).await.into_result()?;
         Ok(r.actual_length())
     }
 
-    async fn usb_reset(&self) -> Result<Self::Reset, Self::Error> {
+    async fn usb_reset(&mut self) -> Result<Self::Reset, Self::Error> {
+        self.interface.take().map(|iface| drop(iface));
```

Comment:
That's a long way to write `self.interface = None` xD I don't mind though

Is there any reason for doing this that way? It's perfectly okay if it's just a personal preference! I'm just curious.